### PR TITLE
Fuse subsections for kdtree tech improvements; add a few refs

### PR DESCRIPTION
**Pull request by tylerjereddy**
Fusing the subsections in kdtree tech improvements section was briefly discussed in the BIDS sprint; also tried to add one or two more refs that might be relevant.

### CODE REVIEW

File: scipy-1.0/ckdtree.tex [hunk 1]
```diff
@@ -45,21 +41,14 @@ \subsubsection{sparse distance matrix}
...
-
-The cKDTree module implements the dual tree counting algorithm in (cite).
-
-An improvement to the pair counting algorithm is added to improve the scale
-with the number of bins.
+The cKDTree module implements a dual tree counting algorithm\cite{pmlr-v28-curtin13},
+with an improvement to the pair counting algorithm to improve the scaling
+with the number of bins. The cKDTree can now be augmented by weights, with 
+weighted paircount essential in many scientific applications, e.g. computing 
+correlation function of galaxies\cite{0004-637X-750-1-38}.
```
Comment (rgommers) [inline scipy-1.0/ckdtree.tex]: textual: "functions" or "the ... function"

File: scipy-1.0/ckdtree.tex [hunk 2]
```diff
@@ -45,21 +41,14 @@ \subsubsection{sparse distance matrix}
...
 data structures.
 
-
-\subsubsection{Pair-Counting}
-
-The cKDTree module implements the dual tree counting algorithm in (cite).
-
-An improvement to the pair counting algorithm is added to improve the scale
-with the number of bins.
+The cKDTree module implements a dual tree counting algorithm\cite{pmlr-v28-curtin13},
```
Comment (rgommers) [inline scipy-1.0/ckdtree.tex]: let's see if one of @rainwoodman or @sturlamolden can confirm whether this is an appropriate reference or not

Comment (tylerjereddy) [reply scipy-1.0/ckdtree.tex]: @rainwoodman [Indicated](https://github.com/scipy/scipy-articles/pull/18#issuecomment-400831802) that:

> Gray and Moore are the dual tree algorithm indeed. 

So I should switch to that reference, which is [here](https://arxiv.org/abs/astro-ph/0012333)

**Comment by rgommers** [comment]
LGTM, merging. Thanks Tyler

**Pull request merged**